### DISCUSSION TITLE
Free Form submission using Axios API using react js

### DISCUSSION THREAD

**Discussion by 360adaptive001** [solspace/craft-freeform#306] [Questions & Answers]
I am creating a website using react js and trying to submit freeform using axios but after submitting I get 400 in response 'Bad Request'
Here is the React Code which I am using to submit the form
<pre>
axios({
      url: 'Site_URL',
      method: 'post',
      data: values,
      headers: {
        'Cache-Control': 'no-cache',
        'X-Requested-With': 'XMLHttpRequest',
        'HTTP_X_REQUESTED_WITH': 'XMLHttpRequest',
        'Access-Control-Allow-Origin': '*',
      },
    })
    //const request = await axios.post(API_ENDPOINT,{data:values})
      .then((response)=>{
        console.log(response);
        
      })
      .catch((error)=>{
        console.log(error);
      
      });
      
     </pre> 
This is the value object I am passing
<pre>
{
    "companyName": "test",
    "firstName": "test",
    "email": "test@gmail.com",
    "handle": "message",
    "formHash": "0Pq3pNb7Q-AqYENRE3b-U0G67rTtHXyIKLjerP46IECA32NvJlOKkXMGT8M9",
    "action": "freeform/submit",
    "CRAFT_CSRF_TOKEN": "vuXD4uX6V09vkNVCurOTCnLcLMrnEs_x6I8FPPzTToCG-BDtmn5J8G7FvhwXlgdAjP0m_2EqrnwUnYjzs8lJKNjRz7bktHXYj827KHd-xcmfp8n9phn9j9rRI7Q4d2UfhJyYq-otAami9KRapmTML-FLPPXeHjIclnir__dd2tqr7n_mHZQY2MFJLja23hBW8rpJ6aeqraicACpC4LsJjNzCchGtSLm0YJyIj91RSbPlGNfvsiDc7AQ4ohLziSoio2Ry6swVywIeyB8nr-nLhKp6GenirdTQhUW7vKuKHUZNofuvk_mcxy7I677pFdAIFFEps6_5z9oZNJqUo-Yz6SmqeIRmTJuVKF1N7hvam4QOqInSiS2yaNsujpYgZAaHqGonoMZ4"
    
}
</pre>

**Comment by gustavs-gutmanis** [comment]
Hi @360adaptive001 

The `Access-Control-Allow-Origin` has to be present in the response, not the request. For that we're making use of the [`allowedGraphqlOrigins`](https://craftcms.com/docs/3.x/config/config-settings.html#allowedgraphqlorigins) craft general configuration entry. If you pass your origin or an asterisk, then it would be set up and used in the response.

```php
// [craft-root]/config/general.php

return [
  // ...
  'allowedGraphqlOrigins' => ['*'], // or ['https://yourdomain.com'] which is better
];
```

This is a very recent and unreleased change though, so you would need to run the `dev-master` version of Freeform until this new update is released.

**Reply by 360adaptive001** [reply]
Thanks for your Quick Reply.
But I think Origin is not the issue When I submit on same domain Form get submitted to the server but In the response I get
400 Bad request.
I am not able to fix the issue.
I am using Axios to submit the form , Form is build with custom html using React Js for frontend development
